File: Python/Algorithms/1346.py
class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        nums = set()
        for num in arr:
            # check if 2 * M in nums
            if num * 2 in nums:
                return True
            # check if M / 2 is in nums
            half = num / 2
            if half % 1 == 0 and half in nums:
                return True
            nums.add(num)
        return False

File: Rust/Algorithms/1346.rs
use std::collections::HashSet;

impl Solution {
    pub fn check_if_exist(arr: Vec<i32>) -> bool {
        let mut nums = HashSet::with_capacity(arr.len());
        for num in arr.into_iter() {
            if nums.contains(&(num * 2)) {
                return true;
            }
            let half = (num as f32) / 2.;
            if half % 1. == 0. && nums.contains(&(half as i32)) {
                return true;
            }
            nums.insert(num);
        }
        return false;
    }
}
